scripts/nayduck.py

```diff
@@ -50,6 +50,11 @@ def _parse_args():
                        '-i',
                        action='store_true',
                        help='Read test set from standard input.')
+    parser.add_argument(
+        '--dry-run',
+        '-n',
+        action='store_true',
+        help='Prints list of tests to execute, without doing anything')
```

Comment:
With https://github.com/near/nearcore/pull/5500 maybe this should be renamed to `-p --print`, `-o --output` or something similar and put in a new mutually exclusive group together with `--run-locally`?

Reply:
@mina86 I think it would make sense to be able to combine those two options

`./scripts/nayduck.py --dry-run --run-locally` should print list of local commands to run.

Reply:
Right, makes sense.

Reply:
@mina86 Hmm, we could rename it to `--output`. However, in that case `--output` name doesn't suggest it stops the execution.

How about, we don't run anything by default, and add an option `--run`?

Reply:
I think your previous suggestion to make `--dru-run --run-locally` print the commands that would be executed makes sense.

Reply:
@mina86 

We have a few options on how to structure the nayduck interface.

Option 1
`./scripts/nayduck.py` // runs remotely
`./scripts/nayduck.py  --run-locally` // runs commands locally
`./scripts/nayduck.py  --dry-run` // prints list of tasks
`./scripts/nayduck.py  --help` // prints help
`./scripts/nayduck.py  --run-locally --dry-run` // print list of local commands

Option 2
```
`./scripts/nayduck.py` // prints help
`./scripts/nayduck.py --run` // runs commands remotely
`./scripts/nayduck.py --run-locally` // runs commands locally
`./scripts/nayduck.py --run --dry-run` // prints list of tasks
`./scripts/nayduck.py --run-locally --dry-run` // print list of local commands
```
where `--run` and `--run-locally` are mutually exclusive

Option 3
```
`./scripts/nayduck.py` // prints help
`./scripts/nayduck.py` --run // runs commands remotely
`./scripts/nayduck.py` --run-locally // runs commands locally
`./scripts/nayduck.py` --dry-run // prints list of tasks without running them
`./scripts/nayduck.py` --dry-run-locally // print list of local commands, that you can just copy paste to run a test
```
where `--run`,  `--run-locally` ,  `--dry-run`, `--print-local-commands` are mutually exclusive

Option 1 is where we will be at once we merge #5500 #5497

I like options 2 and 3. Option 2 may be easier to use, on the other hand option 3 may be easier to explain inside `--help`.
@mina86 What do you think

Reply:
I generally try to avoid changing existing behaviours which is why to me the first option seems to be the best choice.

Reply:
@mina86 That's an excellent point. Option 1 is backward compatible, it's a safe choice.